### PR TITLE
FIX: Load .js files from plugins in qunit testing env

### PR DESCRIPTION
**Pull request by markvanlan**
Files returned from `DiscoursePluginRegistry.each_globbed_asset`  are not filtered by the extension passed into the function!. Then we solved this by filtering out each file with the wrong extension where the function is used by doing this: `elsif f.to_s.end_with?(".#{ext}")`. This doesn't make sense and we should just correct the glob by changing 
`Dir.glob("#{root}/**/*") do |f|` to `Dir.glob("#{root}/**/*.#{ext}") do |f|`

Also JS tests in plugins using `transpile_js`, were not loading the `.js` files.

### CODE REVIEW

File: lib/plugin/instance.rb
Comment (eviltrout) [inline lib/plugin/instance.rb:591]: Isn't this intentionally omitted? Should only be there with `transpile_js`, otherwise plugins that had extra js files in there will have them in the app.

Comment (markvanlan) [reply lib/plugin/instance.rb:591]: It should be `DiscoursePluginRegistry.register_glob(root_path, 'js') if transpile_js` I think.

**Review by eviltrout — APPROVED** [review]
This is cool, but we should be very careful it does not break plugins before the new release. In fact we might want to wait until after the release to merge it.

**Pull request merged**
